src/renderer/Drawable.ts

```diff
@@ -379,4 +383,35 @@ export default class Drawable {
 
     return this._matrix;
   }
+
+  public getSpriteScale(): number {
+    if (this._sprite instanceof Stage) {
+      return 1;
+    } else {
+      const { size } = this._sprite;
+      if (typeof size !== "number") {
+        this._warnBadSize(typeof size, "0%");
+        return 0;
+      } else if (isNaN(size)) {
+        this._warnBadSize("NaN", "0%");
+        return 0;
+      } else if (size === -Infinity) {
+        this._warnBadSize("negative Infinity", "0%");
+        return 0;
+      } else if (size < 0) {
+        this._warnBadSize("less than zero", "0%");
```

Comment:
Are we ever treating an invalid size as something that *isn't* 0%? Will we?

Reply:
No. But to effectively inline the "0%" parameter would dissociate it from the `return 0`. A series of or-conditionals:

```js
if (
  typeof size !== "number" ||
  isNaN(size) ||
  size === -Infinity ||
  size < 0
) {
  ...
  return 0;
}
```

would remove ability to nicely indicate which case occurred (in warning message). Would require some rather nonsense.

```js
let description: string;
if (
  (typeof size !== "number" && description = typeof size) ||
  (isNaN(size) && butWhy = "NaN") ||
  (size === -Infinity && butWhy = "negative Infinity") ||
  (size < 0 && butWhy = "less than zero")
) {
  this._warnBadSize(description); /* or inline */
  return 0;
}
```

Redundancy makes `this._warnBadSize` behavior more declarative where it is read.